iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/fifo.inl

```diff
@@ -24,24 +24,21 @@ namespace iox
 namespace concurrent
 {
 template <class ValueType, uint64_t Capacity>
-inline bool FiFo<ValueType, Capacity>::push(const ValueType& f_param_r) noexcept
+inline bool FiFo<ValueType, Capacity>::push(const ValueType& value) noexcept
 {
     if (is_full())
     {
         return false;
     }
-    else
-    {
-        auto currentWritePos = m_write_pos.load(std::memory_order_relaxed);
-        m_data[currentWritePos % Capacity] = f_param_r;
+    auto currentWritePos = m_write_pos.load(std::memory_order_relaxed);
+    m_data[currentWritePos % Capacity] = value;
 
-        // m_write_pos must be increased after writing the new value otherwise
-        // it is possible that the value is read by pop while it is written.
-        // this fifo is a single producer, single consumer fifo therefore
-        // store is allowed.
-        m_write_pos.store(currentWritePos + 1, std::memory_order_release);
-        return true;
-    }
+    // m_write_pos must be increased after writing the new value otherwise
+    // it is possible that the value is read by pop while it is written.
+    // this fifo is a single producer, single consumer fifo therefore
+    // store is allowed.
+    m_write_pos.store(currentWritePos + 1, std::memory_order_release);
+    return true;
```

Comment:
With this we have to be careful. This is lock free code and without the `else` branch the compiler is able to reorder instructions. I'll look at this next week with a fresh mind

Reply:
@elBoberido I think the as-if rule prohibits a reordering here (except for line 33 but this should be no problem). What do you think?

Reply:
@FerdinandSpitzschnueffler yes, I think you are right and a reordering of line 33 is also not a problem.